### PR TITLE
docs: Update documentation

### PR DESCRIPTION
**Pull request by gismya**
- [x] I have added automatic tests where applicable
- [x] The PR title is suitable as a release note
- [x] The PR contains a description of what has been changed
- [x] The description contains manual test instructions

## Changes

Starts chipping away at #100. Mostly moving the old RTD documentations to markdown and making sure it's referencing modern development workflow and current features in the client. An amalgamation of the old documentation, the python documentation, the work that @jimmycallin did in an [older branch](https://github.com/ftrackhq/ftrack-javascript/tree/docs) and some newly written things.

## Test

Would be great if anyone could check the code examples for issues.

### CODE REVIEW

**Comment by ffMathy** [comment]
This is great! Moving to Markdown is also going to make it easier for me to contribute to the documentation as well.

**Review by ffMathy — APPROVED** [review]
This is amazing! Great work. Can't wait for this to be out so we can more easily contribute to the documentation too.

Minor comments.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,376 @@
...
+```bash
+yarn add @ftrack/api
+```
+
+## Tutorial
+
+The API uses sessions to manage communication with an ftrack server. Create a session that connects to your ftrack server (changing the passed values as appropriate):
+
+```javascript
+const session = new ftrack.Session(
```
[the remark this answers is not in the document]
Comment (lucaas) [reply README.md:43]: It doesn't, no.

It is mainly written for the browser environment, and we wouldn't want it to pick up any envvars by accident during builds.

File: README.md [hunk 2]
```diff
@@ -1,16 +1,376 @@
...
+const response = await session.createComponent(file);
+const component = response[0].data;
+console.debug("Component", component);
+console.debug("ComponentLocation", response[1].data);
+
+console.debug("Component URL: " + session.getComponentUrl(component.id));
+console.debug("Component thumbnail URL: " + session.thumbnailUrl(component.id));
+```
+
+## Handling Events
```
[the remark this answers is not in the document]
Comment (gismya) [reply README.md:107]: It is, a lot of it is an intermediate

[the remark this answers is not in the document]
Comment (gismya) [reply README.md:107]: It is a lot for one page. It's an intermediary step to something a bit more structured. 

The structure and process of the previous documentation made it harder to get going with actually updating the documentation. It's simplifying everything down to just the information so we can work on presentation and process after we have all the important parts in order.

**Review by lucaas — APPROVED** [review]
Nice work! Feel free to merge as-is or address comments first as you wish.

We will iterate on this, but need some basic info in place in the meantime.

File: README.md [hunk 3]
```diff
@@ -1,16 +1,376 @@
...
+Each project returned will be a plain JavaScript object and contain the selected attributes.
+
+The session contains a few other methods besides `query()`, such as `create()`, `update()` and `delete()`. Next up, let’s take a look at combining the query call with an update operation. Since the method return promises, we can chain various asynchronous operations one after the other.
+
+In the example below a specific project is retrieved, and then its status is set to hidden, hiding the project from the UI.
+
+```javascript
+const projectName = "my_project";
+const response = await session.query(
+  "select id from Project where name is " + projectName
```
[the remark this answers is not in the document]
Comment (gismya) [reply README.md:80]: In general yes. But we'd have to add a section on that first, and that will have to come at a later stage.

File: README.md [hunk 4]
```diff
@@ -1,16 +1,376 @@
...
+To connect to the event hub, run `session.eventHub.connect()`. You can also automatically connect the event hub when it is instantiated by providing the option `autoConnectEventHub` when constructing the Session instance:
+
+```javascript
+session = new ftrack.Session(..., { autoConnectEventHub: true });
+session.eventHub.isConnected();
+```
+
+### Error handling
+
+A lot of the methods involve promises that might not always be resolved for various reasons. The examples will not have any error handling to keep them short, but make sure you have proper error handling when using the `EventHub`.
```
[the remark this answers is not in the document]
Comment (gismya) [reply README.md:126]: Yeah, that section was too vague. Removing until we can add a proper section for it.

File: README.md [hunk 5]
```diff
@@ -1,16 +1,376 @@
...
+        applicationId: 'maya'
+    }
+)
+```
+
+### Sending replies
+
+When handling an event it is sometimes useful to be able to send information back to the source of the event. For example, `ftrack.location.request-resolve` would expect a resolved path to be sent back.
+
+You can craft a custom reply event if you want, but an easier way is just to return the appropriate data from your handler. Any value different from *null* or *undefined* will be automatically sent as a reply:
```
[the remark this answers is not in the document]
Comment (gismya) [reply README.md:173]: We actually don't describe `publishReply` currently. I don't know enough about what a believable example would be to write that section, but I've added a short mention of `publishReply` here.

File: README.md [hunk 6]
```diff
@@ -1,16 +1,376 @@
...
+```
+
+### Limitations
+
+The event hub in the JavaScript API has some minor differences and lacks some
+of the features available in the [python counterpart](https://ftrack-python-api.rtd.ftrack.com/en/latest/handling_events.html).
+
+#### Subscription expressions
+
+The JavaScript API currently only support expressions on the format `topic=value` including wildcard support `topic=ftrack.*`, and more complex expressions such as filtering based on event source or data are not supported.
```
[the remark this answers is not in the document]
Comment (gismya) [reply README.md:244]: Yes, wildcard support being supported is mentioned in the paragraph above.

[the remark this answers is not in the document]
Comment (gismya) [reply README.md:238]: Yes, wildcards is explicitly stated as supported. If the wording is confusing, please suggest an alternate wording.

**Review by lucaas — APPROVED** [review]

**Review by jimmycallin — COMMENTED** [review]
Sweet! 

Found one more file we can remove: MANIFEST.in

We should redirect the old documentation URL to this repository as well.

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ffMathy) [reply README.md:234]: Ah. I misread it lol. It's actually really clear. Just a brain fart.

**Comment by gismya** [comment]
> We should redirect the old documentation URL to this repository as well.

Yes. @lucaas is on it.

**Pull request merged**